recOrder/scripts/cli.py

```diff
@@ -50,36 +29,31 @@ def print_reader_info(reader):
     type=click.Choice(["position", "channel", "p", "c"]),
     help="Layers as 'position' ('p') or 'channel' ('c')",
 )
-def view(filename, position=None, layers=None):
+def view(filename, positions=None, layers=None):
     """View a dataset in napari"""
-    print(f"Reading file:\t {filename}")
-
-    # Workaround waveorder #97
-    # Create napari Viewer before other imports
-    v = napari.Viewer()
-    v.close()
-    from waveorder.io import WaveorderReader
+    click.echo(f"Reading file:\t {filename}")
+    print_info(filename)
 
-    reader = WaveorderReader(filename)
-    print_reader_info(reader)
+    reader = read_micromanager(filename)
 
-    if position == ():  # If empty, open all positions
-        position = range(reader.get_num_positions())
-    position = [int(x) for x in position]
+    if positions == ():  # If empty, open all positions
+        positions = range(reader.get_num_positions())
```

Comment:
@ziw-liu I'm trying to find the equivalent of `get_num_positions`. 

Is `sum(1 for _ in dataset.positions())` a (hacky) equivalent? Is there a better way I should do this?

Reply:
I might request your help with making `recorder view` work with v0.4 zarrs. 

Ideally this will let us open with positions or channels as layers, but on my first pass I realized there'll be some tricky logic to make this happen smoothly for all data types. 

I'm also happy to make this a separate issue+PR if you think it should be its own.

Reply:
> Is `sum(1 for _ in dataset.positions())` a (hacky) equivalent? Is there a better way I should do this?
Same speed but a bit more readable:
```py
len(list(dataset.positions()))
```